FoodServiceAPI/Filters/AsyncExceptionFilter.cs

```diff
@@ -0,0 +1,183 @@
+
+            var content = $"An unexpected error has occurred. You can use the following reference id to help us diagnose your problem: {log.ErrorLogId}";
+
+            return new ContentResult
+            {
+                Content = JsonConvert.SerializeObject(content),
+                StatusCode = StatusCodes.Status500InternalServerError,
+                ContentType = "application/json"
+            };
+        }
+
+        private IActionResult GetResult(ServiceException exception)
+        {
+            var responseBody = new ProblemDetails
+            {
+                Type = exception.ErrorCode.ToString(),
+                Title = exception.ErrorCode.GetDescription(),
+                Detail = exception.Detail
+            };
+
+            if (_hostEnvironment.IsDevelopment())
+            {
+                responseBody.Extensions.Add("stackTrace", exception.ToString());
+            }
+
+            var log = new ServiceExceptionLog
+            {
+                ServiceExceptionLogId = Guid.NewGuid().ToString()
+            };
+
+            var exceptionDescription = exception.Message;
+            exceptionDescription = EnsureStringEndsInPeriod(exceptionDescription);
+            exceptionDescription = EscapeJsonForStringFormatInput(exceptionDescription);
+
+            _logger.LogInformation(exception, "ServiceException: {ExceptionDescription} {{{Log}}}", exceptionDescription, log);
+
+            switch (exception.ErrorCode)
+            {
+                case ErrorCode.Forbidden:
+                    responseBody.Status = StatusCodes.Status403Forbidden;
+                    return new ObjectResult(responseBody)
+                    {
+                        StatusCode = responseBody.Status,
+                        ContentTypes = { "application/problem+json", "application/problem+xml" }
+                    };
+                case ErrorCode.InvalidRequest:
+                    responseBody.Status = StatusCodes.Status400BadRequest;
+                    return new BadRequestObjectResult(responseBody)
+                    {
+                        ContentTypes = { "application/problem+json", "application/problem+xml" }
+                    };
+                case ErrorCode.NotFound:
+                    responseBody.Status = StatusCodes.Status404NotFound;
+                    return new NotFoundObjectResult(responseBody)
+                    {
+                        ContentTypes = { "application/problem+json", "application/problem+xml" }
+                    };
+
+                default:
+                    responseBody.Status = StatusCodes.Status500InternalServerError;
+                    return new ObjectResult(responseBody)
+                    {
+                        StatusCode = responseBody.Status,
+                        ContentTypes = { "application/problem+json", "application/problem+xml" }
+                    };
+            }
+        }
```

Comment:
Long Method. Consider breaking down long methods like GetResult(ServiceException exception) into smaller, more manageable private methods. Note that this is not a mandatory change and can be addressed in future improvements.